api/providers/ShipmentProvider.cs

```diff
@@ -51,6 +52,47 @@ public ShipmentProvider(AppDbContext db, IValidator<Shipment> validator) : base(
         return newShipment;
     }
 
+    public override Shipment? Update(Guid id, BaseDTO updateValues)
+    {
+        ShipmentRequest? req = updateValues as ShipmentRequest;
+        if (req == null) throw new ApiFlowException("Could not process update shipment request. Update failed.");
+
+        Shipment? existingShipment = _db.Shipments.Include(o => o.ShipmentItems).FirstOrDefault(o => o.Id == id);
+        if (existingShipment == null) throw new ApiFlowException($"Shipment not found for id '{id}'");
+
+        existingShipment.OrderId = req.OrderId;
+        existingShipment.OrderDate = DateUtil.ToUtcOrNull(req.OrderDate);
+        existingShipment.RequestDate = DateUtil.ToUtcOrNull(req.RequestDate);
+        existingShipment.ShipmentDate = DateUtil.ToUtcOrNull(req.ShipmentDate);
+        existingShipment.SetShipmentType(req.ShipmentType);
+        existingShipment.SetShipmentStatus(req.ShipmentStatus);
+        existingShipment.Notes = req.Notes;
+        existingShipment.CarrierCode = req.CarrierCode;
+        existingShipment.CarrierDescription = req.CarrierDescription;
+        existingShipment.ServiceCode = req.ServiceCode;
+        existingShipment.SetPaymentType(req.PaymentType);
+        existingShipment.SetTransferMode(req.TransferMode);
+        existingShipment.TotalPackageCount = req.TotalPackageCount;
+        existingShipment.TotalPackageWeight = req.TotalPackageWeight;
+        if (req.Items != null)
+        {
+            _db.ShipmentItems.RemoveRange(existingShipment.ShipmentItems);
+
+            existingShipment.ShipmentItems = req.Items.Select(si => new ShipmentItem
+            {
```

Comment:
Overal waar je new ShipmentItem maak moet je dit nog effe erbij zetten
```suggestion
            existingShipment.ShipmentItems = req.Items.Select(si => new ShipmentItem(newInstance=true)
```